akka-actor/src/main/scala/akka/actor/Extension.scala

```diff
@@ -132,5 +132,5 @@ trait ExtensionIdProvider {
   /**
    * Returns the canonical ExtensionId for this Extension
    */
-  def lookup: ExtensionId[_ <: Extension]
+  def lookup(): ExtensionId[_ <: Extension]
```

Comment:
isn't this a source incompatible change for the users implementing this trait (and that is the way to implement an extension)? is it a warning or error even?